internal/pkg/model/schema.go

```diff
@@ -124,15 +120,18 @@ type Agent struct {
 	Active bool           `json:"active"`
 	Agent  *AgentMetadata `json:"agent,omitempty"`
 
-	// API key the Elastic Agent uses to authenticate with elasticsearch
+	// Deprecated. Use Outputs instead. API key the Elastic Agent uses to authenticate with elasticsearch
 	DefaultAPIKey string `json:"default_api_key,omitempty"`
 
-	// Default API Key History
-	DefaultAPIKeyHistory []DefaultAPIKeyHistoryItems `json:"default_api_key_history,omitempty"`
+	// Deprecated. Use Outputs instead. Default API Key History
+	DefaultAPIKeyHistory []ToRetireAPIKeysItems `json:"default_api_key_history,omitempty"`
 
-	// ID of the API key the Elastic Agent uses to authenticate with elasticsearch
+	// Deprecated. Use Outputs instead. ID of the API key the Elastic Agent uses to authenticate with elasticsearch
 	DefaultAPIKeyID string `json:"default_api_key_id,omitempty"`
 
+	// ElasticsearchOutputs is the policy output data for each Elasticsearch output. It maps the output name to its data
+	ElasticsearchOutputs map[string]*PolicyOutput `json:"elasticsearch_outputs,omitempty"`
```

Comment:
Looks like from some of the other comments you had it named `Outputs`. I prefer just the name `Outputs`.

How about using that name and adding an output type to the `PolicyOutput`. That way in the future it we start creating API keys for say logstash or something else the model just works.